connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -366,21 +366,25 @@ private boolean sendRecords() {
                 producer.send(
                     producerRecord,
                     (recordMetadata, e) -> {
-                        if (e != null) {
-                            log.error("{} failed to send record to {}: ", WorkerSourceTask.this, topic, e);
-                            log.debug("{} Failed record: {}", WorkerSourceTask.this, preTransformRecord);
-                            producerSendException.compareAndSet(null, e);
-                        } else {
+                        if (e == null || e instanceof org.apache.kafka.common.errors.RecordTooLargeException) {
                             recordSent(producerRecord);
                             counter.completeRecord();
-                            log.trace("{} Wrote record successfully: topic {} partition {} offset {}",
+                            if(recordMetadata != null)
```

Comment:
Things look good. Happy that the problem will be solved forever :)

One small change, adding braces { } around if and else will make it more readable. That is the standard convention. let us add them.